src/fparser/Fortran2003.py

```diff
@@ -5421,30 +5434,43 @@ class Read_Stmt(StmtBase): # R910
     use_names = ['Io_Control_Spec_List', 'Input_Item_List', 'Format']
     @staticmethod
     def match(string):
-        if string[:4].upper()!='READ': return
+        if string[:4].upper() != 'READ':
+            return
         line = string[4:].lstrip()
         if line.startswith('('):
             line, repmap = string_replace_map(line)
-            i = line.find(')')
-            if i==-1: return
-            l = line[1:i].strip()
-            if not l: return
-            if i==len(line)-1:
-                return Io_Control_Spec_List(repmap(l)),None,None
-            return Io_Control_Spec_List(repmap(l)), None, Input_Item_List(repmap(line[i+1:].lstrip()))
-        if not line: return
-        c = line[0].upper()
-        if 'A'<=c<='Z' or c=='_' or '0'<=c<='9': return
+            idx = line.find(')')
+            if idx == -1:
+                return
+            trimline = line[1:idx].strip()
+            if not trimline:
+                return
+            if idx == len(line) - 1:
+                return Io_Control_Spec_List(repmap(trimline)), None, None
+            return Io_Control_Spec_List(repmap(trimline)), None, \
+                Input_Item_List(repmap(line[idx+1:].lstrip()))
+        if not line:
+            return
+        char = line[0].upper()
+        # No parentheses therefore first argument must be a format
+        # specifier (either a string or a line/lable number
+        if 'A' <= char <= 'Z' or char == '_':
+            return
         line, repmap = string_replace_map(line.lstrip())
-        i = line.find(',')
-        if i==-1: return Format(repmap(line)),None,None
-        l = repmap(line[i+1:].lstrip())
-        if not l: return
-        return None, Format(repmap(line[:i].rstrip())), Output_Item_List(l)
+        # There must be a comma betwee the format specifier and the following
+        # list of values/variables
+        idx = line.find(',')
+        if idx == -1:
+            return None
```

Comment:
You now return None whereas before a format object was returned. Could you explain the reason for the difference please.

Reply:
OK, thanks for explaining.